internal/secretref/secret_ref.go

```diff
@@ -16,51 +17,32 @@ type Getter interface {
 	GetSecretRefs() []telemetryv1alpha1.SecretKeyRef
 }
 
-func ReferencesNonExistentSecret(ctx context.Context, client client.Reader, getter Getter) bool {
-	refs := getter.GetSecretRefs()
-	for _, ref := range refs {
-		hasKey := checkIfSecretHasKey(ctx, client, ref)
-		if !hasKey {
-			return true
-		}
-	}
-
-	return false
-}
+var (
+	ErrSecretKeyNotFound = errors.New("One or more keys in a referenced secret are missing") //nolint:stylecheck //Message will be used in condition message and must be capitalized
+	ErrSecretRefNotFound = errors.New("One or more referenced Secrets are missing")          //nolint:stylecheck //Message will be used in condition message and must be capitalized
+)
 
-func ReferencesSecret(secretName, secretNamespace string, getter Getter) bool {
+func VerifySecretReference(ctx context.Context, client client.Reader, getter Getter) error {
 	refs := getter.GetSecretRefs()
 	for _, ref := range refs {
-		if ref.Name == secretName && ref.Namespace == secretNamespace {
-			return true
+		if _, err := GetValue(ctx, client, ref); err != nil {
+			return err
 		}
 	}
 
-	return false
+	return nil
 }
 
 func GetValue(ctx context.Context, client client.Reader, ref telemetryv1alpha1.SecretKeyRef) ([]byte, error) {
 	var secret corev1.Secret
 	if err := client.Get(ctx, types.NamespacedName{Name: ref.Name, Namespace: ref.Namespace}, &secret); err != nil {
-		return nil, err
+		logf.FromContext(ctx).V(1).Info(fmt.Sprintf("Unable to get secret '%s' from namespace '%s'", ref.Name, ref.Namespace))
+		return nil, fmt.Errorf("%w, first finding is: secret '%s' of namespace '%s'", ErrSecretRefNotFound, ref.Name, ref.Namespace)
```

Comment:
what do you mean by 'first finding is' ?
are there more to come?

Reply:
I improved the wording